helix-core/src/main/java/org/apache/helix/spectator/RoutingDataCache.java

```diff
@@ -84,13 +104,23 @@ public synchronized void refresh(HelixDataAccessor accessor) {
       LOG.info("Reload CurrentStates. Takes " + (System.currentTimeMillis() - start) + " ms");
     }
 
+    if (_sourceDataType.equals(PropertyType.CUSTOMIZEDVIEW) && _propertyDataChangedMap
```

Comment:
With more cases, using if to check _sourceDataType.equals() does not work well.
Can we change to use switch-case?

Reply:
Done. Thanks for the suggestion